src/features/tags.ts

```diff
@@ -60,3 +57,38 @@ export function knownTagsReducer(
 }
 
 export const TagsStore = configureStore({ reducer: knownTagsReducer });
+
+export const addKnownTag = (
+  tagTitle: string,
+  uid: string,
+  item: ItemType,
+  knownTagsLength: number
+) => (dispatch: any) => {
+  dispatch(reload(true));
+  const itemsRef = ref.child(`users/${uid}/knownTags`);
+  const newTagRef = itemsRef.push();
+  const firebaseTag: any = {
+    title: tagTitle,
+    color:
+      COLORS[
+        knownTagsLength + 1 >= COLORS.length
+          ? COLORS.length - 1
+          : knownTagsLength
+      ],
+  };
+  newTagRef.set(firebaseTag, (error) => {
+    if (error) {
+      dispatch(popSnack(true, "error"));
+    } else {
+      const storeKnownTag = {
+        id: newTagRef.key,
+        ...firebaseTag,
+      };
+      dispatch(knownTagAdded(storeKnownTag));
+      dispatch(assignTag(storeKnownTag, item, uid));
+      dispatch(addTab(storeKnownTag, uid));
+      dispatch(popSnack(true, "success"));
```

Comment:
These four consecutive dispatches are imo a nice manifestation why redux or some sort of similar state management tool wins over spaghetti-like approach.